### PR TITLE
Reorganize Jenkins jobs - Subtask #851 - Closes #852

### LINKED ISSUE #852: Reorganisation Jenkins job handling
Parent #851

### PR DESCRIPTION
**Pull request by diego-G**
Besides its main purpose, this PR fixes and checks all the npm scripts. That way we guarantee its working for the issue #794.   

Subtask #851 
Close #852

### CODE REVIEW

File: test/functional/http/post/index.js [hunk 1]
```diff
@@ -0,0 +1,3 @@
+require('./0.transfer');
+require('./1.second.secret');
+require('./2.delegate');
```
Comment (MaciejBaj) [inline test/functional/http/post/index.js]: Missing empty line

File: test/functional/ws/index.js [hunk 2]
```diff
@@ -3,3 +3,4 @@ require('./transport.blocks');
 require('./transport.client');
 require('./transport.handshake');
 require('./transport.transactions');
+require('./transport.transactions.stress');
```
Comment (MaciejBaj) [inline test/functional/ws/index.js]: Missing empty line

File: package.json [hunk 3]
```diff
@@ -94,4 +96,4 @@
     "webpack": "=2.6.1",
     "webpack-node-externals": "=1.6.0"
   }
-}
+}
```
Comment (MaciejBaj) [inline package.json]: Missing empty line

File: Gruntfile.js [hunk 4]
```diff
@@ -161,15 +171,17 @@ module.exports = function (grunt) {
 	grunt.registerTask('jenkins-extensive', ['exec:coverageSingleExtensive']);
 	grunt.registerTask('coverageReport', ['exec:coverageReport']);
 	grunt.registerTask('eslint-nofix', ['eslint']);
-	grunt.registerTask('test', ['eslint', 'exec:coverage']);
```
Comment (MaciejBaj) [inline Gruntfile.js]: Missing most important task: `test`. Also `test-functional`. Please recover them.

File: Gruntfile.js [hunk 5]
```diff
@@ -161,15 +171,17 @@ module.exports = function (grunt) {
...
+	grunt.registerTask('test-functional-http-get-extensive', ['eslint', 'exec:testFunctionalHttpGetExtensive']);
+	grunt.registerTask('test-functional-http-post', ['eslint', 'exec:testFunctionalHttpPost']);
 	grunt.registerTask('test-integration', ['eslint', 'exec:testIntegration']);
 
 	grunt.registerTask('eslint-fix', 'Run eslint and fix formatting', function () {
 		grunt.config.set('eslint.options.fix', true);
 		grunt.task.run('eslint');
 	});
-};
+};
```
Comment (MaciejBaj) [inline Gruntfile.js]: Missing empty line.

File: Gruntfile.js (hunk 4 above)
Comment (diego-G) [reply Gruntfile.js]: `test` script was/is not working anymore. `test-functional` either. They are split it into several scripts.

File: Jenkinsfile [hunk 6]
```diff
@@ -468,4 +389,4 @@ lock(resource: "Lisk-Core-Nodes", inversePrecedence: true) {
 			report()
 		}
 	}
-}
+}
```
Comment (MaciejBaj) [inline Jenkinsfile]: Missing empty line.

File: test/functional/http/get/loader.js [hunk 7]
```diff
@@ -3,31 +3,34 @@
...
+				node.expect(res.body).to.have.property('blocks').to.be.a('number');
+				node.expect(res.body).to.have.property('height').to.be.a('number');
+				node.expect(res.body).to.have.property('broadhash').to.be.a('string');
+				node.expect(res.body).to.not.have.property('consensus'); // Indicates forced forging
+				done();
+			});
 		});
 	});
-});
+});
```
Comment (MaciejBaj) [inline test/functional/http/get/loader.js]: Missing empty line.

File: test/functional/http/get/multisignatures.js [hunk 8]
```diff
@@ -7,133 +7,136 @@ var sendLISK = require('../../../common/apiHelpers.js').sendLISK;
...
+				node.expect(res.body).to.have.property('success');
+				node.expect(res.body).to.have.property('success').to.be.ok;
+				node.expect(res.body).to.have.property('transactions').that.is.an('array');
+				node.expect(res.body.transactions.length).to.equal(0);
+				done();
+			});
 		});
 	});
-});
+});
```
Comment (MaciejBaj) [inline test/functional/http/get/multisignatures.js]: Missing empty line.

File: Gruntfile.js (hunk 4 above)
Comment (MaciejBaj) [reply Gruntfile.js]: How is that? Just chaining grunt task should work just fine:
```
	grunt.registerTask('test-functional', ['test-functional-ws', 'test-functional-http']);
        grunt.registerTask('test', ['test-functional', 'test-unit']);
```

File: test/functional/http/get/loader.js (hunk 7 above)
Comment (LucasIsasmendi) [inline test/functional/http/get/loader.js]: new line

**Comment by diego-G** [comment]
PR corrupted due to unknown reasons. Probably Github issue or VSCode framework. Open PR #871 instead.

**Pull request closed without merging**